phpBB/includes/ucp/ucp_register.php

```diff
@@ -552,7 +549,37 @@ function main($id, $mode)
 
 			'COOKIE_NAME'		=> $config['cookie_name'],
 			'COOKIE_PATH'		=> $config['cookie_path'],
-		));
+		);
+
+		$tpl_name = 'ucp_register';
+		$page_title = 'UCP_REGISTRATION';
+
+		/**
+		* Modify template data on the registration page
+		*
+		* @event core.ucp_register_modify_template_data
+		* @var	array	template_vars		Array with template data
+		* @var	array	data				Array with user data
+		* @var	array	error				Array with errors
+		* @var	string	s_hidden_fields		HTML with hidden form field elements
+		* @var	string	tpl_name			Template name
+		* @var	string	page_title			Page title
+		* @since 3.2.2-RC1
+		*/
+		$vars = array(
+			'template_vars',
+			'data',
+			'error',
+			's_hidden_fields',
```

Comment:
The vars `$data` (except `$data['tz']`), `$error` and `$s_hidden_fields` were already used to build `$template_vars`. This means overwriting them in an event doesn't have any effect. 

To fix this you could split up creation of `$template_vars` before and after the event.

Reply:
Thanks for checking this!

> This means overwriting them in an event doesn't have any effect.

Yes, but reading them can still be useful. Extensions will have to recreate the `$template_vars` value though. If it's split up, extensions don't have a chance to modify the final template value.

Reply:
@kasimi Yes they are useful, but should be marked as readonly in the event comments if you can't change them. However when I look at the final template output of `$error` and `$s_hidden_fields` I think these assignments can just be moved after the event so they're editable:

```
'ERROR'				=> (sizeof($error)) ? implode('<br />', $error) : '',
```
```
'S_HIDDEN_FIELDS'	=> $s_hidden_fields,
```

When doing it like this you can easily add or remove errors and you're also able to replace the text of an error without having to implode() and (re)parse to the template again. It's also easy to add an additional hidden field.

The `$data` variable is more tricky. I first thought of just marking that as "readonly" in the event comments, but  `$data['tz']` is being used after the event, meaning the value of it can be edited in the event. (so can't mark it as readonly) When looking at the final template values, the only value that differs from the actual $data value, is this assignment:
```
'S_LANG_OPTIONS'	=> language_select($data['lang']),
```
If we move this assignment to after the event, you can indeed no longer change the final template value of it.

So what we could do is mark `$data` as readonly and create a separate `$tz` variable (from`$data['tz']`) and pass that to the event as being editable. If you have a better way, let me know!

Reply:
* The call to `phpbb_timezone_select()` has been moved above the event. Subscribers will need to clear the `timezone_select` and `timezone_date` template blocks prior to calling this function again, if they intend to.

* The `$event['data']` array  has been marked _read-only_.

* The `$event['error']` array is imploded and added to the `$template_vars` array _below_ the event. This allows subscribers to easily manipulate the array without having to re-assign it to `$event['template_vars']`.

* The `$event['s_hidden_fields']` array is added to the `$template_vars` array _below_ the event for the same reason. Also, the call to `build_hidden_fields()` has been moved below the event which now receives the form elements in an array.